partner_firstname/res_user.py

```diff
@@ -35,8 +35,15 @@ def create(self, cr, user, vals, context=None):
         installed
         """
         vals2 = vals.copy()
+
         if 'name' in vals:
             vals2['lastname'] = vals2['name']
+        elif 'lastname' not in vals and 'partner_id' in vals:
+            res_partner = self.pool.get('res.partner')
+            partner = res_partner.browse(cr, user, vals2['partner_id'],
+                                         context)
+            if partner.lastname:
```

Comment:
Can it actually be the case that the partner has no lastname? In that case, you need to refactor the code below to also assign the login to vals['lastname'] if there is a partner but it does not have a lastname. This clause is now skipped as soon as there is a partner_id.